fetchai/ledger/api/common.py

```diff
@@ -253,3 +271,62 @@ def _post_tx_json(self, tx_data: bytes, endpoint: Optional[str]):
         tx_list = response.get('txs', [])
         if len(tx_list):
             return tx_list[0]
+
+    def submit_signed_tx(self, tx: Transaction, signatures: Dict[Identity, bytes]):
+        """
+        Appends signatures to a transaction and submits it, returning the transaction digest
+        :param tx: A pre-assembled transaction
+        :param signatures: A dict of signers signatures
+        :return: The digest of the submitted transaction
+        :raises: ApiError on any failures
+        """
+        # Encode transaction and append signatures
+        encoded_tx = transaction.encode_multisig_transaction(tx, signatures)
+
+        # Submit and return digest
+        return self._post_tx_json(encoded_tx, tx.action)
+
+
+class TransactionFactory:
+    API_PREFIX = None
+
+    @classmethod
+    def _create_skeleton_tx(cls, fee: int):
+        # build up the basic transaction information
+        tx = Transaction()
+        tx.charge_rate = 1
+        tx.charge_limit = fee
+
+        return tx
+
+    @classmethod
+    def _create_action_tx(cls, fee: int, entity: AddressLike, action: str, shard_mask: Optional[BitVector] = None):
+        tx = cls._create_skeleton_tx(fee)
+        tx.from_address = Address(entity)
+        tx.target_chain_code(cls.API_PREFIX, shard_mask if shard_mask else BitVector())
+        tx.action = action
+        return tx
+
+    @classmethod
+    def _encode_json(cls, obj):
+        return json.dumps(obj).encode('ascii')
+
+    @classmethod
+    def _encode_msgpack_payload(cls, *args):
```

Comment:
This method _encode_msgpack_payload is still retained elsewhere and is now a duplicate. Since requires and holds no state refactor this.